knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -613,6 +613,13 @@ func configureEventingKafka(spec serverlessoperatorv1alpha1.KnativeKafkaSpec) mf
 			return scheme.Scheme.Convert(deployment, u, nil)
 		}
 
+		if u.GetKind() == "ConfigMap" && u.GetName() == "kafka-config-logging" {
+			log.Info("Found ConfigMap kafka-config-logging, updating it with values from spec")
+			if err := unstructured.SetNestedField(u.Object, renderLoggingConfigXML(spec.Logging.Level), "data", "config.xml"); err != nil {
```

Comment:
This is potentially rendering an empty string

Reply:
Doing some default, like for ha, in here: https://github.com/openshift-knative/serverless-operator/pull/1765/files#diff-3b74c6630d5f7a1fc62c6f999d5c1bc3572e5b8603ad46edc46c51dce4eab0adR296

Reply:
@matzew Link you posted is recursive :D

I guess you mention https://github.com/openshift-knative/serverless-operator/pull/1765/files#diff-8a3b1481924d4139a463a9c68ccdaa1056686b4209ab0be3ef82b488127b8f97R127-R140

I assumed the specification in the CRD yaml that does the validation and defaulting.

Are there any cases where the created custom object escapes Kube api server validation @pierDipi ? Or, don't those validations and defaulting specified in OpenAPI definitions are not working the way we think they would?

Reply:
> Are there any cases where the created custom object escapes Kube api server validation @pierDipi ?

That's exactly my original question, so we should try and see what happens, for reference:
- https://github.com/kubernetes/apiextensions-apiserver/issues/28
- https://github.com/kubernetes/kubernetes/issues/62325#issuecomment-444767470

Reply:
We default it to `INFO` on the serverside,see the code. And if I change the value to some bogus value, the openapi validates does reject the change.

IMO this is all fine, as is now